src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.cs

```diff
@@ -216,7 +216,7 @@ public override IEnumerable<NodeSmoProperty> SmoProperties
                 properties.Add(new NodeSmoProperty
                 {
                    Name = "IsLedger",
-                   ValidFor = ValidForFlag.AllOnPrem
+                   ValidFor = ValidForFlag.Sql2022
```

Comment:
This means we're going to have to update this every time a new version comes out which is going to be a pain to maintain. I would suggest adding a "Sql2022OrHigher" value to the enum so that we only have to update it in one place when a new version is added. 

(another option would be to do like SMO does and add a "MinValidForFlag" that takes in a single value and returns true if the server version is that or higher. This has the advantage of not having to remember to update the "OrHigher" values when a new version is added. But the first option seems more straightforward and that extra bit of overhead shouldn't be too bad.

Reply:
I think with #1802 we should be able to address that.

Reply:
Probably, but I wouldn't suggest relying on that being done anytime soon so better to add in the enum value now (it's only a single line added and then update value in xml) so that we don't forget and end up paying for it down the road when the next version is released.

Reply:
Are there examples of properties that are only valid for a certain subset of SQL versions? Like valid in a previous version but not a later version? I took a quick look through the xml and didn't see any like that. 

This issue with the ValidFor enums came up when I was making changes for Ledger; it was a pain adding "Sql2022" everywhere. I think ideally there'd be a "Sql20xxOrHigher" for each SQL version, so the only place that would need to be updated for future editions would be the ValidFor enum definition. #1802 is the cleanest long-term solution though, I agree.

Reply:
I just added more flags to streamline the pattern.. please check again @nofield :)

Reply:
In the future we may need to support features that are only valid for a certain range, but currently that's not something we support so I think it's fine to just go with the minimum required here and then #1802 can cover special scenarios like that in a much easier way.